components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/OIDCSessionManagerTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.oidc.session;
+
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.oidc.session.cache.OIDCSessionParticipantCache;
+import org.wso2.carbon.identity.oidc.session.servlet.TestOIDCSessionBase;
+import org.wso2.carbon.identity.oidc.session.servlet.TestUtil;
+
+import static org.testng.Assert.*;
+
+/**
+ * Unit test coverage for OIDCSessionManager
+ */
+@PrepareForTest({OIDCSessionParticipantCache.class})
+public class OIDCSessionManagerTest extends TestOIDCSessionBase {
+
+    @Mock
+    OIDCSessionState oidcSessionState;
+
+    private OIDCSessionManager oidcSessionManager;
+    private static final String SESSION_ID = "090907ce-eab0-40d2-a46d-acd4bb33f0d0";
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        oidcSessionManager=new OIDCSessionManager();
+        TestUtil.startTenantFlow("carbon.super");
```

Comment:
Do you always need to have a tenant flow started before all tests ? If not only start the tenant flow whenever required.

Reply:
Yes. I need